src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.SimplifyTypeNames
+{
+    internal class TypeSyntaxSimplifierWalker : CSharpSyntaxWalker
+    {
+        private readonly CSharpSimplifyTypeNamesDiagnosticAnalyzer _analyzer;
+        private readonly SemanticModel _semanticModel;
+        private readonly OptionSet _optionSet;
+        private readonly CancellationToken _cancellationToken;
+
+        public List<Diagnostic> Diagnostics { get; } = new List<Diagnostic>();
+
+        public TypeSyntaxSimplifierWalker(CSharpSimplifyTypeNamesDiagnosticAnalyzer analyzer, SemanticModel semanticModel, OptionSet optionSet, CancellationToken cancellationToken)
+            : base(SyntaxWalkerDepth.StructuredTrivia)
+        {
+            _analyzer = analyzer;
+            _semanticModel = semanticModel;
+            _optionSet = optionSet;
+            _cancellationToken = cancellationToken;
+        }
+
+        public override void VisitQualifiedName(QualifiedNameSyntax node)
+        {
+            if (node.IsKind(SyntaxKind.QualifiedName) && TrySimplify(node))
```

Comment:
the IsKind check isn't necessary, right?

Reply:
I believe some of the tests can be removed, but I wanted the initial refactoring to be explicit.

Reply:
k :)